packages/common/src/hooks/useUSDCBalance.ts

```diff
@@ -31,22 +31,24 @@ export const useUSDCBalance = () => {
     [dispatch]
   )
 
-  useEffect(() => {
-    const fetch = async () => {
-      setStatus(Status.LOADING)
-      try {
-        const account = await getUserbankAccountInfo(audiusBackend, {
-          mint: 'usdc'
-        })
-        const balance = (account?.amount ?? new BN(0)) as BNUSDC
-        setData(balance)
-        setStatus(Status.SUCCESS)
-      } catch (e) {
-        setStatus(Status.ERROR)
-      }
+  const refresh = useCallback(async () => {
+    if (status === Status.LOADING) return
+    setStatus(Status.LOADING)
+    try {
+      const account = await getUserbankAccountInfo(audiusBackend, {
+        mint: 'usdc'
+      })
+      const balance = (account?.amount ?? new BN(0)) as BNUSDC
+      setData(balance)
+      setStatus(Status.SUCCESS)
+    } catch (e) {
+      setStatus(Status.ERROR)
     }
-    fetch()
   }, [audiusBackend, setData])
```

Comment:
One thing that occurs to me here:

In the mobile drawer, we use the balance fetching status to conditionally delay rendering of the form until we've loaded it. This means that if the status flips back to LOADING, the form is going to disappear arbitrarily and be replaced with a spinner.

We could either:
* Find a way to incorporate the recovery status into the overall modal loading status so that we check for recovery _first_ and then only show the modal once recovery has finished (might significantly delay the loading time if we have to retry)
* I can update the logic in the forms to not care about the loading state of the balance, which would cause the value to just magically change once the balance has loaded. This might be a jarring experience. It's worth getting some feedback from design on what the preferred pattern is.

Reply:
the way this works in web, the value will just update if the recovery flow does happen. basically: `magically change`. this should be a super rare occurrence anyway so it's probably fine. also okay to discuss with design if you have strong opinions.
this can def be its own experience, with another modal (or sthg else) showing the recovery flow happening, etc... just like the buy audio flow. but i think we were trying to avoid building all that for this edge case. cc @rickyrombo

Reply:
I think we start with this and if we see a lot of recoveries we can smooth out the UI, but this should be really really rare